src/snowflake/snowpark/session.py

```diff
@@ -227,9 +237,14 @@ def configs(
 
         def create(self) -> "Session":
             """Creates a new Session."""
-            if "connection" in self._options:
-                return self._create_internal(self._options["connection"])
-            return self._create_internal(conn=None)
+            session = self._create_internal(self._options.get("connection"))
+            for k, v in self._options.items():
+                if k not in (
+                    "password",
+                    "connection",
+                ):  # Parameters we choose not to expose in RuntimeConfig
+                    session.conf._conf[k] = v
```

Comment:
Why don't we initiate `conf` here directly with a dict that contains the exposed config?